File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
n = int(input())

for i in range(n):
    x = int(input())
    dp = [0] * (x + 1)

    for j in range(1, x + 1):
        if (j == 1):
            dp[j] = 1
        elif (j == 2):
            dp[j] = 2
        elif (j == 3):
            dp[j] = 4
        else:
            dp[j] = dp[j - 1] + dp[j - 2] + dp[j - 3]

    print(dp[x])

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let lines = [];
rl.on("line", (input) => {
  lines.push(input.trim());
});

rl.on("close", () => {
  const T = Number(lines[0]);

  let dp = Array.from({ length: 11 }, () => 0);
  dp[1] = 1;
  dp[2] = 2;
  dp[3] = 4;

  for (let i = 4; i < 11; i++) {
    dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3];
  }

  for (let i = 0; i < T; i++) {
    const n = Number(lines[i + 1]);
    console.log(dp[n]);
  }
});
